ncbi_genome_download/core.py

```diff
@@ -185,56 +252,98 @@ def download(**kwargs):
         to use multiprocessing for requests
     table : str
         file to store metadata
+    Returns
+    -------
+    int
+        success code
+    """
+    # Parse and pre-process keyword arguments
+    parser=argument_parser()
+    group = kwargs.pop('group', EDefaults.TAXONOMIC_GROUPS.default)
+    args = parser.parse_args([group])
+
+    args.section = kwargs.pop('section', EDefaults.SECTIONS.default)
+    args.uri = kwargs.pop('uri', EDefaults.URI.default)
+    args.output = kwargs.pop('output', EDefaults.OUTPUT.default)
+    args.file_format = kwargs.pop('file_format', EDefaults.FORMATS.default)
+    args.assembly_level = kwargs.pop('assembly_level', EDefaults.ASSEMBLY_LEVELS.default)
+    args.refseq_category = kwargs.pop('refseq_category', EDefaults.REFSEQ_CATEGORIES.default)
+    args.genus = kwargs.pop('genus', EDefaults.GENUS.default)
+    args.species_taxid = kwargs.pop('species_taxid', EDefaults.SPECIES_TAXID.default)
+    args.taxid = kwargs.pop('taxid', EDefaults.TAXID.default)
+    args.human_readable = kwargs.pop('human_readable', False)
+    args.parallel = kwargs.pop('parallel', EDefaults.NB_PROCESSES.default)
+    args.metadata_table = kwargs.pop('metadata_table', EDefaults.TABLE.default)
+    assert len(kwargs) == 0, "Unrecognized option(s): {}".format(kwargs.keys())
 
+    return args_download(args)
+
+def args_download(args):
+    """
+    Download data from NCBI
+    Using the argument parser object.
+
+    Parameters
+    ----------
+    args: Namespace
+        Arguments from argument_parser
+_
     Returns
     -------
     int
         success code
 
     """
     # Parse and pre-process keyword arguments
-    section = kwargs.pop('section', EDefaults.SECTIONS.default)
-    assert section in EDefaults.SECTIONS.choices, "Unsupported section: {}".format(section)
-    group = kwargs.pop('group', EDefaults.TAXONOMIC_GROUPS.default)
-    assert group in EDefaults.TAXONOMIC_GROUPS.choices, "Unsupported group: {}".format(group)
-    if group == 'all':
+    assert args.section in EDefaults.SECTIONS.choices, "Unsupported section: {}".format(args.section)
+    for group in args.group.split(','):
```

Comment:
It would of course be nice to not re-split `group` here, just like in the code below.